app/models/hint.rb

```diff
@@ -1,5 +1,36 @@
+require 'stringex/core_ext'
+
 class Hint < ApplicationRecord
   def self.match(searchterm)
-    Hint.find_by(fingerprint: searchterm)
+    searchprint = fingerprint(searchterm)
+    Hint.find_by(fingerprint: searchprint)
   end
+
+  # This implements the OpenRefine fingerprinting algorithm. See
+  # https://github.com/OpenRefine/OpenRefine/wiki/Clustering-In-Depth#fingerprint .
```

Comment:
rubocop will stop yelling about this if you remove the trailing ` .` from the url. It'll still be too long, but it's
reasonable smart about not freaking out about stuff like this.